04_numeronym/undrewb/main_test.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"bytes"
+	"reflect"
+	"strconv"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	want := strconv.Quote("[a11y K8s abc]\n")
+	got := strconv.Quote(buf.String())
+
+	if got != want {
+		t.Errorf("actual: %s does not match expected: %s", got, want)
+	}
+}
+
+var numeronymsData = []struct {
+	name  string
+	input []string
+	want  []string
+}{
+	{name: "lab example",
+		input: []string{"accessibility", "Kubernetes", "abc"},
+		want:  []string{"a11y", "K8s", "abc"}},
+	{name: "empty",
+		input: []string{},
+		want:  []string{}},
+}
+
+func TestNumeronyms(t *testing.T) {
```

Comment:
I have a couple of suggestions for tests for you to consider:

- How would the functions behave for test data involving unicode strings "Hello, 世界"
- Since TestNumeronym and TestNumeronyms share similar test cases(other than the type of the input parameter), the tests could be combined into one function that calls both methods. I know this is not always favourable but in this case the tests might be more readable. Please ignore if you do not want to.

Reply:
excellent pickup! It didnt cope with unicode strings. Ive updated my implementation and added test cases.